### ISSUE TITLE
KtLint gives wrong autoCorrect flag to custom rule.

### ISSUE THREAD

**Issue by MyDogTom** [shyiko/ktlint#154]
**Overview**
My custom rule always receives `autoCorrect = false`, regardless that I'm calling `emit` with `canBeAutocorrected = true`.

**Description**
My custom rule is a copy of `IndentationRule`, but with enabled autocorrection. It means that both rules detect same error in same place. `IndentationRule` emits error with `canBeAutocorrected = false`, my `CustomIndentationRule`  emits error with `canBeAutocorrected = true`

**How to reproduce.**
I have repository which show case the problem. https://github.com/MyDogTom/ktlint-custom-rules/tree/fix-format (branch `fix-format`)
It has intentional [indent violation](https://github.com/MyDogTom/ktlint-custom-rules/blob/fix-format/src/main/kotlin/com/github/mydogtom/ktlint/rules/CustomIndentationRule.kt#L57).
Auto correction works only when I remove `autoCorrect` check. See [this commit](https://github.com/MyDogTom/ktlint-custom-rules/commit/cac949235e92043cb3f62c0d5680de95aadc457b#diff-8698d5f747fefe08c12e4aa678cdaf7aL81)

For sure, it somehow conflicts with standard `IndentionRule`, but it shouldn't. Unfortunately, I couldn't quickly figure out where is problem.

**Comment by MyDogTom** [comment]
As experiment, I removed `IndentionRule` from `StandardRuleSetProvider`. Behavior stayed the same. Autocorrection happens only when I remove `if (autoCorrect)` from `CustomIndentationRule`. At least bug it's not related to the fact that one rule is a copy of another. I still cannot find why that happens. Any help is really welcome :)

**Comment by shyiko** [comment]
```sh
$ ktlint --version
0.15.0

$ cat A.kt 
fun main(
    args: String) {
println("")
}

$ ktlint -R com.github.MyDogTom:ktlint-custom-rules:master-SNAPSHOT A.kt -v
/tmp/A.kt:2:17: Parenthesis should be on new line (custom:class-and-function-header-format)

$ ktlint -R com.github.MyDogTom:ktlint-custom-rules:master-SNAPSHOT A.kt -F
<no output>

$ ktlint -R com.github.MyDogTom:ktlint-custom-rules:master-SNAPSHOT A.kt
<no output>
```
Looks like everything is OK (at least in ktlint@0.15.0).
Can you provide an example with a different behavior?

**Comment by MyDogTom** [comment]
Sure.
```kotlin
fun main(args: String) {
   println("")
}
```

`ktlint -R com.github.MyDogTom:ktlint-custom-rules:master-SNAPSHOT A.kt -v`
>A.kt:2:1: Unexpected indentation (3) (it should be 4) (indent)
>A.kt:2:1: Unexpected indentation (3) (it should be 4) (custom:indent)

`ktlint -R com.github.MyDogTom:ktlint-custom-rules:master-SNAPSHOT A.kt -F`
>A.kt:2:1: Unexpected indentation (3) (it should be 4)

`ktlint -R com.github.MyDogTom:ktlint-custom-rules:master-SNAPSHOT A.kt -v`
>A.kt:2:1: Unexpected indentation (3) (it should be 4) (indent)
>A.kt:2:1: Unexpected indentation (3) (it should be 4) (custom:indent)

**Comment by shyiko** [comment]
Fix available in [0.15.1](https://github.com/shyiko/ktlint/blob/0.15.1/CHANGELOG.md#0151---2018-02-14) (Maven Central sync might take up to 25 minutes). :balloon:

**Comment by MyDogTom** [comment]
Thank you for quick fix. I've checked new version via `brew` and it works! Unfortunately, new version is still not available via Maven/JCenter.

**Comment by shyiko** [comment]
@MyDogTom thank you for letting me know. https://issues.sonatype.org/browse/OSSRH-37894 submitted.

**Comment by shyiko** [comment]
@MyDogTom it looks like Maven Central sync finally took place

**Comment by MyDogTom** [comment]
Thank you!
